docs/production/authentication-methods.md

```diff
@@ -420,7 +420,12 @@ it as follows:
     3. (Optional) Put the Zulip server public certificate in `/etc/zulip/saml/zulip-cert.crt`
        and the corresponding private key in `/etc/zulip/saml/zulip-private-key.key`. Note that
        the certificate should be the single X.509 certificate for the server, not a full chain of
-       trust, which consists of multiple certificates.
+       trust, which consists of multiple certificates. The private key cannot be encrypted
+       with a password, as then Zulip will not be able to load it. An example pair can be
+       generated using:
+       ```
+       openssl req -x509 -newkey rsa:2056 -keyout zulip-private-key.key -out zulip-cert.crt -days 365 -nodes
```

Comment:
Why 2056 rather than 2048?

Reply:
Oh yeah, I was meaning to ask that too and got distracted.  I would have expected `2048` as well.

Reply:
Hmm I went with what https://www.samltool.com/self_signed_certs.php has - 2056. Though  I don't know why they do that so maybe we should switch to the more natural 2048...

Reply:
https://developers.onelogin.com/saml/online-tools/x509-certs/obtain-self-signed-certs also offers 1024 and 2056 in their dropdown, so it does feel like this is a SAML thing.